cmd/glasskube/cmd/bootstrap_gitops.go

```diff
@@ -0,0 +1,126 @@
+			cliutils.ExitWithError()
+		}
+
+		currentContext := color.New(color.Bold).Sprint(clicontext.RawConfigFromContext(ctx).CurrentContext)
+		fmt.Fprintf(os.Stderr, "Glasskube and ArgoCD will be installed in context %s.\n", currentContext)
+		if !bootstrapCmdOptions.yes && !cliutils.YesNoPrompt("Continue?", true) {
+			cancel()
+		}
+
+		// regularly bootstrap in the cluster
+		// without a git client, this means it only supports GitHub for now
+		basePath := strings.ReplaceAll(bootstrapGitopsCmdOptions.repo, "github.com", "raw.githubusercontent.com")
+		basePath = basePath + "/main"
+		manifestsPath := fmt.Sprintf("%v/bootstrap/glasskube/manifests.yaml", basePath)
+		_, err := bootstrapClient.Bootstrap(ctx, bootstrap.BootstrapOptions{
+			Url:        manifestsPath,
+			Type:       bootstrap.BootstrapTypeAio,
+			GitopsMode: true,
+			Force:      true,
+		})
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "\nAn error occurred during bootstrap:\n%v\n", err)
+			cliutils.ExitWithError()
+		}
+
+		cliutils.SetupClientContext(true, util.Pointer(true))(cmd, make([]string, 0))
+
+		// install argo-cd package
+		// TODO use the version of argo given in /packages/argo-cd/clusterpackage.yaml instead
+		argoCdPkg := client.PackageBuilder("argo-cd").
+			WithRepositoryName("glasskube").
+			WithVersion("v2.11.5+1").
+			BuildClusterPackage()
+
+		if _, err := install.NewInstaller(cliutils.PackageClient(cmd.Context())).
+			InstallBlocking(cmd.Context(), argoCdPkg, metav1.CreateOptions{}); err != nil {
+			fmt.Fprintf(os.Stderr, "\nAn error occurred installing argo-cd package:\n%v\n", err)
+			cliutils.ExitWithError()
+		}
+
+		fmt.Fprintf(os.Stderr, "argo-cd package has been installed\n")
+
+		// apply bootstrap/glasskube-application.yaml into the cluster
+		appPath := fmt.Sprintf("%v/bootstrap/glasskube-application.yaml", basePath)
+		if objs, err := clientutils.FetchResources(appPath); err != nil {
+			fmt.Fprintf(os.Stderr, "\nAn error occurred fetching the bootstrap application:\n%v\n", err)
+			cliutils.ExitWithError()
+		} else if client, err := dynamic.NewForConfig(cfg); err != nil {
+			fmt.Fprintf(os.Stderr, "\nAn error occurred initializing the dynamic client:\n%v\n", err)
+			cliutils.ExitWithError()
+		} else {
+			for _, obj := range objs {
+				if _, err = client.Resource(schema.GroupVersionResource{
+					Group:    "argoproj.io",
+					Version:  "v1alpha1",
+					Resource: "applications",
+				}).Namespace(obj.GetNamespace()).Apply(ctx, obj.GetName(), &obj, metav1.ApplyOptions{
+					FieldManager: "glasskube", // TODO not sure if correct
+				}); err != nil {
+					fmt.Fprintf(os.Stderr, "\nAn error occurred applying the bootstrap application:\n%v\n", err)
+					cliutils.ExitWithError()
+				}
+			}
+		}
+
+		// wait for argo to be ready, then print...
+		// TODO
+
+		fmt.Fprintf(os.Stderr, "DONE\n")
+
+	},
+}
+
+func init() {
+	bootstrapGitopsCmd.Flags().StringVar(&bootstrapGitopsCmdOptions.repo, "repo", "", "URL of the GitOps Repository")
```

Comment:
Not sure about the conventions, but since `--url` is required anyway, wouldn't it make sense that it is an argument instead? `glasskube bootstrap git https://...`

Reply:
as @kosmoz said Flux is doing it similarly and we should define the parameter as required. It just gives us the option to add more parameter / subcommands in the future :-)